extension/src/inpage/provider.ts

```diff
@@ -244,13 +251,19 @@ export class IronProvider extends SafeEventEmitter {
    * if provided and marks this provider as initialized.
    * Throws if called more than once.
    */
-  protected async initializeStateAsync() {
+  protected async initialize() {
```

Comment:
Is it possible to have concurrent calls to `initialize`? Asking because you're only setting `this.state.initialized` at the end of the function.

Reply:
only setting at the end because it's fallible. need to make sure the request succeeds before assuming it's initialized (e.g.: if the app is not even running)

if you check all the `handle*` functions, they already avoid emitting duplicate events if the previous state is the same, so this should be safe